### DISCUSSION TITLE
PCE 6button can not work.

### DISCUSSION THREAD

**Discussion by alexgame8818** [darthcloud/BlueRetro#396] [Q&A]
Thanks to your project again.
I am working on PCE BlueRetro,but 6button in StreetfighterII can not work correctly.
I also tested some FW,like 0.16,1.22,1.41, which fixed for PCE.
2button is OK all the time.
Before FW1.41,6button mode can be recognized as a 1P-6button.But can not work correctly.
From FW1.41,6button mode will be recognized as a 2P-2button which can work correctly ,and 1P-6button can be seen in game option mode which can not be controlled. Of cause I was just using one controller for firt connection to BlueRetro.  
But at this time,start key will work for 1P and 2P at the same time.
So after entering the game,I can control 2P by 2button mode,and 1P was always keeping squat.
While pressing a key, 1P and 2P will show different action at the same time.
I also checked the key mapping,that's ok as well as other console's BlueRtro.

I want to ask if there is anyone has this problem,maybe I will try to rebuild it again by a new ESP32.

**Comment by darthcloud** [comment]
Make sure you are well grounded (using the connector shield (housing), not just the GND pin)

Double check that the OE and SEL are not inversed and that they got both a 47K pull-up.

PCE 6btns is very sensitive timing wise.

1.7.3 is known to work fine.

**Reply by alexgame8818** [reply]
It really works! Just by connecting the shield's GND.
OE and SEL with 2 47K was completly followed the Cables Build Instructions.
But because I found GND pin was not connected to the shield on original PCE 2 button pad.
I stumped for three days...thanks a lot.

And I want to ask another question.
The Slow(turbo pause) and keeping shooting(in shootting games) need the Autofire locked.
Is that possible to set a key as a Switch to turn on and off the autofire lock.

**Reply by darthcloud** [reply]
Not really the best is to map the button twice, one with turbo and the other without. Btw HumbleBazooka is the one that figured the GND problem!

**Reply by alexgame8818** [reply]
Thanks to HumbleBazooka too.
Maybe he found the original 6button pad‘s GND pin connected to the shield, which 2button pad didnt care :D

About autofire, I mean, just like some HORI sticks have a switch for the keys to 3 modes-normal, turbo and keeping turbo.
So PS4&PS5 controller has enough keys for NES and PCE, is that possible to set a key likes L1R1 as a swtich between normal A button and keeping A button autofire.

**Reply by darthcloud** [reply]
I understand what you means, but BlueRetro doesn't work like that.

Best you can do is map a second A button on R1 for example with turbo set for that one.

**Reply by alexgame8818** [reply]
For NES,I am trying to set START with turbo on L1. It seems more interesting that we can control the SLOW in game likes a super skill :D

Thanks for so much help. I will try to build more Blueretros on more consoles.
